### PR TITLE
Update hash to take into account Swift version & add tests (fixes #13)

### LINKED ISSUE #13: Hash should take into account Xcode version
Since Swift frameworks [aren't binary compatible between releases](https://developer.apple.com/swift/blog/?id=2), cached frameworks currently break when developers are on different versions of Xcode or if everyone doesn't upgrade at once. As a workaround, `carthage_cache` could just take into account `xcodebuild -version` when generating the file name for each cache.

(This happened to us today with the release of Xcode 7.3.)

### PR DESCRIPTION
**Pull request by msanders**
See #13.

### CODE REVIEW

File: spec/carthage_cache/archiver_spec.rb [hunk 1]
```diff
@@ -10,7 +10,7 @@
   describe "#archive" do
 
     it "creates a zip file with the content of the project's 'Carthage/Build' directory" do
-      expected_command = "cd #{build_directory} && zip -r -X #{archive_path} iOS Mac tvOS watchOS > /dev/null"
+      expected_command = "cd #{build_directory} && zip -r -X #{archive_path}  > /dev/null"
```
Comment (guidomb) [inline spec/carthage_cache/archiver_spec.rb]: Why are you changing this?

**Comment by guidomb** [comment]
@msanders I would avoid having to pass the executor from the `Application` down to every object until the `CartfileResolvedFile`. I would suggest to make a class like `SwiftVersionResolver` (which could use the executor). Then the `CartfileResolvedFile` could depend on this swift version resolver (providing a default instance). Did I make myself clear? What do you think?

```ruby
require "digest"

module CarthageCache

  class CartfileResolvedFile

    attr_reader :file_path, :swift_version_resolver

    def initialize(file_path, swift_version_resolver = SwiftVersionResolver.new)
      @file_path = file_path
      @swift_version_resolver = swift_version_resolver
    end

    def digest
      @digest ||= Digest::SHA256.hexdigest(content + "#{swift_version}")
    end

    def content
      @content ||= File.read(file_path)
    end

    def swift_version
      @swift_version ||= swift_version_resolver.swift_version
    end

  end

end
```

**Comment by msanders** [comment]
@guidomb That sounds cleaner, I'll try it out.

File: spec/carthage_cache/archiver_spec.rb (hunk 1 above)
Comment (msanders) [inline spec/carthage_cache/archiver_spec.rb]: It was breaking the spec, but the change you outlined below should hopefully fix it.

**Comment by msanders** [comment]
@guidomb Updated per your feedback

**Comment by guidomb** [comment]
@msanders Thanks!

**Pull request merged**